src/views/blog-post/table-of-contents/mobile/table-of-contents-mobile.module.scss

```diff
@@ -0,0 +1,294 @@
+
+	--toc-mobile_item_min-height: var(--min-target-size_m);
+	--toc-mobile_item_padding-start: var(--spc-4x);
+	--toc-mobile_item_padding-end: var(--spc-4x);
+	--toc-mobile_item_padding-vertical: var(--spc-2x);
+
+	--toc-mobile_sub-item_padding-start: var(--spc-3x);
+	--toc-mobile_sub-item_dot_margin-end: var(--spc-3x);
+	--toc-mobile_sub-item_dot_size: 1rem;
+
+	--toc-mobile_label_color: var(--foreground_emphasis-medium);
+	--toc-mobile_label_color_selected: var(--foreground_emphasis-high);
+
+	--toc-mobile_item_background-color_hovered: var(--surface_primary_emphasis-medium);
+	--toc-mobile_item_background-color_pressed: var(--surface_primary_emphasis-high);
+	--toc-mobile_item_background-color_selected: var(--surface_primary_emphasis-low);
+	--toc-mobile_item_background-color_focused: var(--background_focus);
+
+	--toc-mobile_subheading-dot_color: var(--primary_default);
+
+	--toc-mobile_focus-outline_color: var(--focus-outline_primary);
+	--toc-mobile_focus-outline_width: var(--border-width_focus);
+
+	--toc-mobile_divider_width: var(--border-width_s);
+	// --background_disabled applied to background
+	--toc-mobile_divider_color: #cddbe7;
+
+	--toc-mobile_arrow_color: var(--foreground_emphasis-high);
+	--toc-mobile_arrow_color_selected: var(--primary);
+
+	@include until($tabletLarge) {
+		--toc-mobile_list_padding-horizontal: var(--spc-2x);
+	}
+
+	
+	@include until($mobile) {
+		--toc-mobile_list_padding-horizontal: 0px;
+
+		--toc-mobile_header_padding-top: 0px;
+
+		--toc-mobile_header_padding-horizontal: var(--site-spacing);
+		--toc-mobile_header_padding-bottom: 0px;
+	}
+
+}
+
+.mobileTocContainer {
+
+	@include darkTheme() {
+		--toc-mobile_divider_color: #253e4c;
+	}
+
+	top: 0;
+	max-height: calc(100vh - ((var(--header_logo_size) + var(--header_padding-vertical) *2 ) ) );
+
+	.tocTitle {
+
+	
+		width: 100%;
+		background-color: var(--toc-mobile-header_background-color);
+		margin: 0;
+		padding: 0 var(--toc-mobile_header_padding-horizontal);
+		border: 1px solid var(--background_disabled);
+		border-width: 1px 0 1px 0;
+		display: flex;
+		align-items: center;
+		justify-content: space-between;
+
+		h4 {
+			white-space: nowrap;
+			overflow: hidden;
+			text-overflow: ellipsis;
+			width: 90%;
+		}
+
+		button {
+			.mobileTocOpenIcon { display: none; }
+			.mobileTocCloseIcon { display: inline;  }
+		}
+	
+	}
+	
+	.tableList {
+	
+		background-color: var(--toc-mobile-header_background-color);
+		list-style: none;
+		margin: 0;
+		padding: var(--toc-mobile_list_padding-vertival) var(--toc-mobile_list_padding-horizontal);
+		height: calc(100vh - ((var(--header_logo_size) + var(--header_padding-vertical) *2 ) ) - ( 1.5rem + (16.0 * 2px)));
+		overflow: auto;
+		transition: height 0.50s;
+	
+	}
+
+	&.tocMobileMenuCollapsed {
+
+		.tocTitle {
+			button {
+				.mobileTocOpenIcon { display: inline; }
+				.mobileTocCloseIcon { display: none;  }
+			}
+		}
+
+		.tableList {
+			height: 0px;
+			padding: 0;
```

Comment:
This `padding: 0` means that the content abruptly shifts to zero padding when closing the menu. The scrollbar also causes layout shift when it appears/disappears during this animation (depends on the browser; yours may or may not have any effect on the layout). I think these are both limitations of animating by `height`

What I'd suggest here is actually animating between `clip-path: inset(0)` (fully visible) and `clip-path: inset(0 0 100% 0)` (fully hidden) instead - this way no layout changes occur during the animation. You'll also need to set `position: absolute;` on the `.tableList` for this to work.

Reply:
https://github.com/josephmfaulkner/playfulprogramming/commit/76faae41d6cfad4c7348dc92b05b505750b1bf80